prow/config/github/oauth.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package config
```

Comment:
Could we not make these rename/move changes? This complicates the diff substantially.

Reply:
I have to because there would be dependency cycle otherwise:

```
prow/flagutil
  prow/githuboauth
    prow/config
      pod-utils/decorate
        ..
          prow/flagutil
```

Reply:
Seems like the auth config structures can just live in the flagutil package (and/or pkg/ghclient or prow/github). Do not see a need for a new package here.

Reply:
OK, let me try flagutil.

Reply:
flagutil won't work:
```
% bazel build //prow:release                                                                                                                                                                                                                                                                                                                           !1217
ERROR: /mnt/gopath/src/k8s.io/test-infra/prow/flagutil/BUILD.bazel:3:1: in go_library rule //prow/flagutil:go_default_library: cycle in dependency graph:
    //prow:release
    //robots/commenter:image
    //robots/commenter:app
    //robots/commenter:app.binary
    //robots/commenter:go_default_library
.-> //prow/flagutil:go_default_library
|   //prow/githuboauth:go_default_library
`-- //prow/flagutil:go_default_library
This cycle occurred because of a configuration option
ERROR: Analysis of target '//prow:release' failed; build aborted
INFO: Elapsed time: 7.714s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (829 packages loaded, 7005 targets configured)
```
Trying `pkg/ghclient` and `prow/github` now.

Reply:
`pkg/ghclient` is a wrapper around go-github, and `prow/github` provides all the interfaces on github API. Neither of them are good fit in my opinion.

I actually feel `prow/githuboauth` is the best place to move to.

Reply:
The package move has been done in the second commit, I'll squash if that's acceptable.